camel/models/base_model.py

```diff
@@ -30,25 +32,75 @@ def __init__(self, model_type: ModelType,
             model_config_dict (Dict[str, Any]): A config dictionary.
         """
         self.model_type = model_type
+
         self.model_config_dict = model_config_dict
+        self.check_model_config()
+
+        self._token_counter: Optional[BaseTokenCounter] = None
```

Comment:
Move this to inheritants

Reply:
@Obs01ete Yeah I followed your suggestion and moved the definition `self._token_counter: Optional[BaseTokenCounter] = None` into the inheritants. But it seems each of the inheritant will repeat the same statement, does this matter? 🤔

Reply:
It is fine in this case